gnss_lib_py/utils/gnss_models.py

```diff
@@ -0,0 +1,658 @@
+            except KeyError:
+                warnings.warn("Assuming 0 "+ row + " for Rx", RuntimeWarning)
+                state[row] = 0
+
+        # Compute measurements
+        if pseudorange or doppler:
+            est_meas, sv_posvel = expected_measures(gps_millis, state,
+                                                    ephem=rx_ephem,
+                                                    sv_posvel=sv_posvel)
+            # Reverse the sorting to match the input measurements
+            est_meas = est_meas.sort(ind=inv_sort_order)
+        else:
+            est_meas = None
+        if corrections:
+            est_clk, est_trp, est_iono = calculate_pseudorange_corr(gps_millis,
+                                        state=state, ephem=rx_ephem,
+                                        sv_posvel=sv_posvel, iono_params=iono_params)
+            # Reverse the sorting to match the input measurements
+            est_clk = est_clk[inv_sort_order]
+            est_trp = est_trp[inv_sort_order]
+            est_iono = est_iono[inv_sort_order]
+        else:
+            est_clk = None
+            est_trp = None
+            est_iono = None
+        # Add required values to new rows
+        if sv_posvel is not None:
+            # Reverse the sorting to match the input measurements
+            sv_posvel = sv_posvel.sort(ind=sorted_sats_ind)
+        est_frame = NavData()
+        if pseudorange:
+            est_frame['est_pr_m'] = est_meas['est_pr_m']
+        if doppler:
+            est_frame['est_doppler_hz'] = est_meas['est_doppler_hz']
+        if corrections:
+            est_frame['b_sv_m'] = est_clk
+            est_frame['tropo_delay_m'] = est_trp
+            est_frame['iono_delay_m'] = est_iono
+        if use_posvel:
+            for row in sv_state_rows:
+                est_frame[row] = sv_posvel[row]
+        if len(est_measurements)==0:
+            est_measurements = est_frame
+        else:
+            est_measurements.concat(est_frame, inplace=True)
+    est_measurements = measurements.concat(est_measurements, axis=0, inplace=False)
+    return est_measurements
+
+
+def simulate_measures(gps_millis, state, noise_dict=None, ephem=None,
+                      sv_posvel=None, rng=None, el_mask=5.):
+    """Simulate GNSS pseudoranges and doppler measurements given receiver state.
+
+    Measurements are simulated by finding satellites visible from the
+    given position (in state), computing the expected pseudorange and
+    doppler measurements for those satellites, corresponding to the given
+    state, and adding Gaussian noise to these expected measurements.
+
+    Parameters
+    ----------
+    gps_millis : int
+        Time at which measurements are needed, measured in milliseconds
+        since start of GPS epoch [ms].
+    state : gnss_lib_py.parsers.navdata.NavData
+        NavData instance containing state i.e. 3D position, 3D velocity,
```

Comment:
I'd suggest adding specifically what row names `state` needs to include. Some variant of `x_rx_*_m` etc.? Not clear what rows actually need to be included in `state`